Note:
In the above video, the null hypothesis is mistakenly represented as 'less than': μ∘ < 2.5. This is not correct as we do not use strict inequalities in the null hypothesis.
Please note that this will be 'less than or equal to': μ∘ ≤ 2.5.

The formulation of the null and alternate hypotheses determines the type of the test and the position of the critical regions in the normal distribution.


You can tell the type of the test and the position of the critical region on the basis of the ‘sign’ in the alternate hypothesis.



       ≠ in H₁    →   Two-tailed test        →     Rejection region on both sides of distribution

       < in H₁    →   Lower-tailed test     →     Rejection region on left side of distribution

       > in H₁    →   Upper-tailed test     →     Rejection region on right side of distribution

A manufacturer claims that the average life of its product is 36 months. An auditor selects a sample of 49 units of the product, and calculates the average life to be 34.5 months. The population standard deviation is 4 months. Test the manufacturer’s claim at 3% significance level using the critical value method.


In [1]:
sigma = 4 / 7

In [2]:
sigma

0.5714285714285714

In [3]:
import math

math.sqrt(49)

7.0

In [4]:
Z_score = -2.17
mu = 34.5

In [5]:
LCV = mu + Z_score * sigma

In [6]:
LCV

33.26

In [7]:
0.03 / 2

0.015

In [8]:
1 - 0.015

0.985

In [9]:
UCV = mu - Z_score * sigma

In [10]:
UCV

35.74

In [11]:
36 * 0.97

34.92

In [12]:
36 + 2.17 * (4 / 7)

37.24

In [13]:
36 - 2.17 * (4 / 7)

34.76

The UCV and LCV values for this test are 37.24 and 34.76. The sample mean in this case is 34.5 months, which is less than LCV. So, this implies that the sample mean lies in the critical region and you can reject the null hypothesis.

Critical Value Method
Consider this problem, H₀: μ ≤ 350 and H₁: μ > 350

So, the Zc comes out to be 1.645. Now, find the critical value for the given Zc and make the decision to accept or reject the null hypothesis.

μ = 350     σ = 90       N (Sample size) = 36
¯
x
= 370.16




In [14]:
mu = 350
sigma = 15
Z = 1.645

In [15]:
mu + Z * sigma

374.675

0.950 is not there in the z-table. So, look for the numbers nearest to 0.950. You can see that the z-score for 0.9495 is 1.64 (1.6 on the horizontal bar and 0.04 on the vertical bar), and the z-score for 0.9505 is 1.65. So, taking the average of these two, the z-score for 0.9500 is 1.645.

Government regulatory bodies have specified that the maximum permissible amount of lead in any food product is 2.5 parts per million or 2.5 ppm. Let’s say you are an analyst working at the food regulatory body of India FSSAI. Suppose you take 100 random samples of Sunshine from the market and have them tested for the amount of lead. The mean lead content turns out to be 2.6 ppm with a standard deviation of 0.6.


One thing you can notice here is that the standard deviation of the sample is given as 0.6, instead of the population’s standard deviation. In such a case, you can approximate the population’s standard deviation to the sample’s standard deviation, which is 0.6 in this case.


Answer the following questions in order to find out if a regulatory alarm should be raised against Sunshine or not, at 3% significance level.

In [16]:
μ = 2.5
σ = 0.6 / 10

In [17]:
Ζ = 1.88

In [18]:
μ + Ζ * σ

2.6128

The critical value can be calculated from μ + Zc x (σ/​√N​), as 2.5 + 1.88(0.6 /​√100​) = 2.61 ppm . You need to use the + sign since the critical value is on the right-hand side (upper-tailed test). Since the sample mean 2.6 ppm is less than the critical value (2.61 ppm), you fail to reject the null hypothesis and don’t raise a regulatory alarm against Sunshine.

In [19]:
2.5 + 1.88 * (0.6 / 10)

2.6128

Summary

So what did you learn in this session? 

1. Hypothesis — a claim or an assumption that you make about one or more population parameters
2. Types of hypothesis:
* Null hypothesis (H₀) - Makes an assumption about the status quo
                                     - Always contains the symbols ‘=’, ‘≤’ or ‘≥’
* Alternate hypothesis (H₁) - Challenges and complements the null hypothesis
                                                                  - Always contains the symbols ‘≠’, ‘<’ or ‘>’
1. Types of tests:
* Two-tailed test - The critical region lies on both sides of the distribution
                             - The alternate hypothesis contains the ≠ sign
* Lower-tailed test - The critical region lies on the left side of the distribution
                                - The alternate hypothesis contains the < sign
* Upper-tailed test - The critical region lies on the right side of the distribution
                                                 - The alternate hypothesis contains the > sign
1. Making a decision - Critical value method:
* Calculate the value of Zc from the given value of α (significance level)
* Calculate the critical values (UCV and LCV) from the value of Zc
* Make the decision on the basis of the value of the sample mean ¯x with respect to the critical values (UCV AND LCV)


Summary

So what did you learn in this session?

Hypothesis — a claim or an assumption that you make about one or more population parameters

Types of hypothesis:

Null hypothesis (H₀) - Makes an assumption about the status quo
- Always contains the symbols ‘=’, ‘≤’ or ‘≥’

Alternate hypothesis (H₁) - Challenges and complements the null hypothesis

- Always contains the symbols ‘≠’, ‘<’ or ‘>’

Types of tests:
Two-tailed test - The critical region lies on both sides of the distribution
- The alternate hypothesis contains the ≠ sign
Lower-tailed test - The critical region lies on the left side of the distribution
- The alternate hypothesis contains the < sign
Upper-tailed test - The critical region lies on the right side of the distribution

- The alternate hypothesis contains the > sign

Making a decision - Critical value method:

Calculate the value of Zc from the given value of α (significance level)

Calculate the critical values (UCV and LCV) from the value of Zc

Make the decision on the basis of the value of the sample mean ¯x with respect to the critical values (UCV AND LCV)

In [20]:
mu

350

In [21]:
mu = 62.6

In [22]:
sigma = 10.7

In [23]:
μ = 62.6
σ = 10.7
sample_size = 100

In [24]:
Z = 2.33

In [25]:
UCV, LCV = μ + Z * (σ / 10), μ - Z * (σ / 10)

In [26]:
LCV, UCV

(60.1069, 65.0931)

In [27]:
62.6 + 2.33 * 1.07

65.0931

In [28]:
62.6 - 2.33 * 1.07

60.1069

In [29]:
60 - 2.33 * 1.07

57.5069

In [30]:
62.6 + 2.33 * 1.7

66.561

In [31]:
62.6 - 2.33 * 1.07

60.1069

In [32]:
62.6 + 2.33 * 1.07

65.0931

In [33]:
60 - 2.33 * 1.07

57.5069

In [34]:
60 + 2.33 * 1.07

62.4931

In [35]:
10.7 / 10

1.0699999999999998

Prof Tricha has defined the p-value as the probability of the null hypothesis being accepted (or more aptly, not being rejected). This statement is not technically correct (or formal) definition of p-value, but it is used for better understanding of the p-value.


Higher the p-value, higher is the probability of failing to reject a null hypothesis. On the other hand, lower the p-value, higher is the probability of the null hypothesis being rejected.


fter formulating the null and alternate hypotheses, the steps to follow in order to make a decision using the p-value method are as follows:


    Calculate the value of z-score for the sample mean point on the distribution

    Calculate the p-value from the cumulative probability for the given z-score using the z-table

    Make a decision on the basis of the p-value (multiply it by 2 for a two-tailed test) with respect to the given value of α (significance value).

To find the correct p-value from the z-score, first find the cumulative probability by simply looking at the z-table, which gives you the area under the curve till that point.




Let’s solve the following problem stepwise to consolidate your learning on how to make a decision about any hypothesis using the p-value method.

You are working as a data analyst at an auditing firm. A manufacturer claims that the average life of its product is 36 months. An auditor selects a sample of 49 units of the product, and calculates the average life to be 34.5 months. The population standard deviation is 4 months. Test the manufacturer’s claim at 3% significance level using the p-value method.

First, formulate the hypotheses for this two-tailed test, which would be:

                                   H₀: μ = 36 months and H₁: μ ≠ 36 months

Now, you need to follow the three steps to find the p-value and make a decision.

Try out the three-step process by answering the following questions. 

You have learnt how to perform the three steps of the p-value method with the help of the AC sales problem as well as the above product lifecycle comprehension problem.

In [36]:
import numpy as np
from scipy import stats

In [46]:
p_value = 0.0044 * 2

In [47]:
p_value

0.0088

def zeta(x, q=None, out=None):
    r"""
    Riemann or Hurwitz zeta function.

    Parameters
    ----------
    x : array_like of float or complex.
        Input data
    q : array_like of float, optional
        Input data, must be real.  Defaults to Riemann zeta. When `q` is
        ``None``, complex inputs `x` are supported. If `q` is not ``None``,
        then currently only real inputs `x` with ``x >= 1`` are supported,
        even when ``q = 1.0`` (corresponding to the Riemann zeta function).

    out : ndarray, optional
        Output array for the computed values.

    Returns
    -------
    out : array_like
        Values of zeta(x).

    See Also
    --------
    zetac

    Notes
    -----
    The two-argument version is the Hurwitz zeta function

    .. math::

        \zeta(x, q) = \sum_{k=0}^{\infty} \frac{1}{(k + q)^x};

    see [dlmf]_ for details. The Riemann zeta function corresponds to
    the case when ``q = 1``.

    For complex inputs with ``q = None``, points with
    ``abs(z.imag) > 1e9`` and ``0 <= abs(z.real) < 2.5`` are currently not
    supported due to slow convergence causing excessive runtime.

    References
    ----------
    .. [dlmf] NIST, Digital Library of Mathematical Functions,
        https://dlmf.nist.gov/25.11#i

    Examples
    --------
    >>> import numpy as np
    >>> from scipy.special import zeta, polygamma, factorial

    Some specific values:

    >>> zeta(2), np.pi**2/6
    (1.6449340668482266, 1.6449340668482264)

    >>> zeta(4), np.pi**4/90
    (1.0823232337111381, 1.082323233711138)

    First nontrivial zero:

    >>> zeta(0.5 + 14.134725141734695j)
    0 + 0j

    Relation to the `polygamma` function:

    >>> m = 3
    >>> x = 1.25
    >>> polygamma(m, x)
    array(2.782144009188397)
    >>> (-1)**(m+1) * factorial(m) * zeta(m+1, x)
    2.7821440091883969

    """
    if q is None:
        return _ufuncs._riemann_zeta(x, out)
    else:
        return _ufuncs._zeta(x, q, out)



In [38]:
from scipy.special import zeta, polygamma, factorial

In [39]:
zeta(2), np.pi**2 / 6

(np.float64(1.6449340668482264), 1.6449340668482264)

In [41]:
np.pi**2 / 6

1.6449340668482264

In [42]:
zeta(2)

np.float64(1.6449340668482264)

You can calculate the z-score for sample mean 34.5 months using the formula: (​​¯x​ - μ) / (σ/​√n) . This gives you (34.5 - 36) / (4/√49) = (-1.5) * 7/4 = -2.62. Notice that, since the sample mean lies on the left side of the hypothesised mean of 36 months, the z-score comes out to be negative.

In [44]:
1 - 0.02 * 0.02

0.9996

The value in the z-table corresponding to -2.6 on the vertical axis and 0.02 on the horizontal axis is 0.0044. Since the sample mean is on the left side of the distribution and this is a two-tailed test, the p-value would be 2 * 0.0044 = 0.0088.

In [45]:
34.5 + 2.62 * (4 / 7)

35.997142857142855

Let’s say you work at a pharmaceutical company that manufactures an antipyretic drug in tablet form, with paracetamol as the active ingredient. An antipyretic drug reduces fever. The amount of paracetamol deemed safe by the drug regulatory authorities is 500 mg. If the value of paracetamol is too low, it will make the drug ineffective and become a quality issue for your company. On the other hand, a value that is too high would become a serious regulatory issue.


There are 10 identical manufacturing lines in the pharma plant, each of which produces approximately 10,000 tablets per hour.


Your task is to take a few samples, measure the amount of paracetamol in them, and test the hypothesis that the manufacturing process is running successfully, i.e. the paracetamol content is within regulation. You have the time and resources to take about 900 sample tablets and measure the paracetamol content in each.


Upon sampling 900 tablets, you get an average content of 510 mg with a standard deviation of 110. What does the test suggest, if you set the significance level at 5%? Should you be happy with the manufacturing process or should you ask the production team to alter the process? Is it a regulatory alarm or a quality issue?


Solve the following questions in order to find out the answers to the questions stated above.


One thing you can notice here is that the standard deviation of the sample of 900 is given as 110, instead of the population’s standard deviation. In such a case, you can use the sample standard deviation (110 in this case) to calculate an approximate population standard deviation.

In [52]:
from math import sqrt

mu = 510
sigma = 110
n = 900
α = 0.05

In [51]:
Z = -1.645

In [53]:
mu + Z * sigma / sqrt(n)

503.96833333333336

In [54]:
mu - Z * sigma / sqrt(n)

516.0316666666666

In [58]:
round((1 - 0.9968) * 2, 5)

0.0064

You can calculate the z-score for the sample mean 510 mg using the formula: (​¯x​ - μ) / (σ /​√N​). This gives you (510 - 500) / (110 /√900) = (10) / (110 / 30) = 2.73. Notice that, since the sample mean lies on the right side of the hypothesised mean of 500 mg, the z-score comes out to be positive.

The value in the z-table corresponding to 2.7 on the vertical axis and 0.03 on the horizontal axis is 0.9968. Since the sample mean is on the right side of the distribution and this is a two-tailed test (because we want to test if the value of the paracetamol is too low or too high), the p-value would be 2 * (1 - 0.9968) = 2 * 0.0032 = 0.0064.

In [59]:
import scipy

scipy.stats.norm.ppf(0.01)

np.float64(-2.3263478740408408)

In [60]:
scipy.stats.norm.ppf(0.001)

np.float64(-3.090232306167813)

In [61]:
from scipy import stats

In [65]:
[stats.t.ppf(1 - 0.005, x) for x in range(1, 31)]

[np.float64(63.65674116287399),
 np.float64(9.92484320091807),
 np.float64(5.840909309733352),
 np.float64(4.604094871415897),
 np.float64(4.032142983557535),
 np.float64(3.707428021324907),
 np.float64(3.4994832973505026),
 np.float64(3.3553873313333957),
 np.float64(3.2498355415921254),
 np.float64(3.16927267261695),
 np.float64(3.10580651553928),
 np.float64(3.0545395893929017),
 np.float64(3.0122758387165773),
 np.float64(2.9768427343708344),
 np.float64(2.946712883485951),
 np.float64(2.920781622496036),
 np.float64(2.8982305196347173),
 np.float64(2.878440472713585),
 np.float64(2.860934606449914),
 np.float64(2.845339709776814),
 np.float64(2.831359558017186),
 np.float64(2.818756060596369),
 np.float64(2.8073356837675227),
 np.float64(2.796939504772804),
 np.float64(2.787435813675851),
 np.float64(2.7787145333289134),
 np.float64(2.7706829571216756),
 np.float64(2.763262455461066),
 np.float64(2.756385903670335),
 np.float64(2.7499956535670305)]

In [ ]:
stats.t.sf()

In [68]:
np.random.seed(42)
sample_data = np.random.normal(loc=77, scale=10, size=250)
population_mean = 75

In [69]:
sample_data

array([ 81.96714153,  75.61735699,  83.47688538,  92.23029856,
        74.65846625,  74.65863043,  92.79212816,  84.67434729,
        72.30525614,  82.42560044,  72.36582307,  72.34270246,
        79.41962272,  57.86719755,  59.75082167,  71.37712471,
        66.8716888 ,  80.14247333,  67.91975924,  62.87696299,
        91.65648769,  74.742237  ,  77.67528205,  62.75251814,
        71.55617275,  78.1092259 ,  65.49006423,  80.75698018,
        70.9936131 ,  74.0830625 ,  70.98293388,  95.52278185,
        76.86502775,  66.42289071,  85.22544912,  64.7915635 ,
        79.08863595,  57.40329876,  63.71813951,  78.96861236,
        84.3846658 ,  78.71368281,  75.84351718,  73.98896304,
        62.2147801 ,  69.80155792,  72.39361229,  87.57122226,
        80.4361829 ,  59.36959845,  80.24083969,  73.1491772 ,
        70.23078   ,  83.11676289,  87.30999522,  86.31280119,
        68.60782477,  73.90787624,  80.31263431,  86.75545127,
        72.20825762,  75.14341023,  65.93665026,  65.03

In [73]:
np.std(sample_data, ddof=2)

np.float64(9.679994252886154)

In [84]:
stats.t.ppf(1 - 0.025, 32)

np.float64(2.036933343460101)